client/src/modules/auth/index.js

```diff
@@ -0,0 +1,111 @@
+//  tada ** request must have body credentials
+//  todo * research axios interceptor
+//  todo * wrap axios inside custom "fetcher"
+//
+
+class ConnectedLoginPage extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            /** user input of corresponding fields */
+            credentials: {
+                //todo STOPSHIP for testing purposes , default values must be empty strings
+                login: "levayv@mail.ru",
+                password: "123456789",
+            }
+        };
+        /** user input inside LOGIN field */
+        this.updateLogin = (event) => {
+            const newLoginValue = event.target.value;
+            this.setState(state =>
+                Object.assign({}, state, {
+                    credentials: {
+                        login: newLoginValue,
+                        password: state.credentials.password,
+                    }
+                }));
+        };
+        /** user input inside PASSWORD field */
+        this.updatePassword = (event) => {
+            const newPasswordValue = event.target.value;
+            this.setState(state =>
+                Object.assign({}, state, {
+                    credentials: Object.assign({}, state.credentials, {
+                            // login: state.credentials.login,
+                            password: newPasswordValue,
+                        }
+                    )
+                }));
+        };
+        /** user click on LOGIN button */
+        this.handleLogin = () => {
+            this.props.loginAttempt(this.state.credentials);    // axios.post(
+        }
+    }
+
+    // todo for testing only
+    // componentDidUpdate(prevProps, prevState, snapshot) {
+    //     console.log(" --- Component Did Update --- ")
+    //     console.log("current / previous state")
+    //     console.log(this.state)
+    //     console.log(prevState)
+    //     console.log("current / previous state")
+    //     console.log(this.props)
+    //     console.log(prevProps)
+    //     console.log(" --- END ---")
+    // }
+
+    render() {
+        return (
+            <div>
+                <h1>Login</h1>
+                <LabeledInput
+                    displayName={"Login"}
+                    placeholder={"YourEmail@mail.xx"}
+                    onChange={this.updateLogin}
+                    value={this.state.credentials.login}
+                />
+                <LabeledInput
+                    displayName={"Password"}
+                    placeholder={"1234"}
+                    onChange={this.updatePassword}
+                    value={this.state.credentials.password}
+                />
+                <Button
+                    displayName={"Sign-in"}
+                    onClick={this.handleLogin}
+                />
+            </div>
+        );
+    }
+}
+
+const mapStateToProps = state => {
+    return {
+        token: state.token,
```

Comment:
need to talk about this

Reply:
I forgot to delete mapStateToProps function which was used for testing token's existence inside  redux store, now it is redundant. Login page doesn't need token during request.
Removed. https://github.com/Levayv/ipma_v2/commit/e59e42e0f616b2de8d0c48f3f1c311eea0ab8a53